tests/System.Buffers.Primitives.Tests/BufferSegment.cs

```diff
@@ -3,13 +3,14 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace System.Buffers.Testing
 {
-    internal class BufferSegment : ReadOnlySequenceSegment<byte>
+    internal class BufferSegment<T> : ReadOnlySequenceSegment<T> where T : struct
```

Comment:
What's the distinction between T : struct and T : unmanaged? I am assuming unmanaged is stricter and means T can't be a referencing containing struct. Should BufferReader\<T\> be constrained to T : struct? Can it be?

Reply:
Unmanaged is blittable. You can take sizeof(), pointers, etc. I figured starting there was better- might give flexibility.

Reply:
We don't need the constraint if we don't use it anywhere.

Reply:
I just added the first usage to BufferReader. I determine whether or not to allocate on the stack based on size(T).